### PR TITLE
supporting new proxy fields into the AlertmanagerConfig CRD for the O…

### PR DESCRIPTION
**Pull request by Tanc010**
…Auth2 clients

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

https://github.com/prometheus-operator/prometheus-operator/pull/6831#pullrequestreview-2250872359

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -1569,12 +1569,44 @@ func (cb *configBuilder) convertHTTPConfig(ctx context.Context, in *monitoringv1
...
+		out.ProxyFromEnvironment = *in.ProxyFromEnvironment
+	}
+
+	if in.ProxyConnectHeader != nil {
+		proxyConnectHeader := make(map[string][]string, len(in.ProxyConnectHeader))
+		s := cb.store.ForNamespace(crKey.Namespace)
+		for k, v := range in.ProxyConnectHeader {
+			proxyConnectHeader[k] = []string{}
+			for _, vv := range v {
+				value, _ := s.GetSecretKey(vv)
```
Comment (dongjiang1989) [inline pkg/alertmanager/amcfg.go:1600]: Maybe check error if get secret nil

File: pkg/alertmanager/amcfg.go [hunk 2]
```diff
@@ -1569,12 +1569,44 @@ func (cb *configBuilder) convertHTTPConfig(ctx context.Context, in *monitoringv1
 			Scopes:         in.OAuth2.Scopes,
 			TokenURL:       in.OAuth2.TokenURL,
 			EndpointParams: in.OAuth2.EndpointParams,
+			proxyConfig:    cb.convertProxyConfig(in.OAuth2.ProxyConfig, crKey),
```
Comment (dongjiang1989) [inline pkg/alertmanager/amcfg.go:1572]: CRD auto define it

**Comment by dongjiang1989** [comment]
cc @simonpasquier PTAL

**Comment by Tanc010** [comment]
cc @simonpasquier  PTAL, thanks

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**